src/urdb.cpp

```diff
@@ -384,102 +380,84 @@ bool OpenEI::QueryUtilityRates(const wxString &name, std::vector<RateInfo> &rate
 	url.Replace("<GUID>", "");
 	url.Replace("<APIKEY>", wxString(sam_api_key));
 
-	json_data = MyGet(url);
-	if (json_data.IsEmpty())
-	{
-		if (err) *err = "Web API call to urdb_rates returned empty JSON for utility company name = " + name; // do not report url because it has private API key
-		return false;
-	}
-// change from UTF8 to UTF16 encoding to address unicode characters per SAM GitHub issue 1848
-	rapidjson::GenericDocument < rapidjson::UTF16<> > reader;
-	reader.Parse(json_data.c_str());
 
-	if (reader.HasParseError())
-	{
-		if (err) *err = wxString::Format("Could not parse JSON for utility company name = %s.\nRapidJSON ParseErrorCode = %d", name, reader.GetParseError());
-			return false;
-	}
 
-	if (reader.HasMember(L"items")) {
-		if (reader[L"items"].IsArray()) {
+	// change from UTF8 to UTF16 encoding to address unicode characters per SAM GitHub issue 1848
+	rapidjson::GenericDocument < rapidjson::UTF16<> > reader;
+	bool retrieve_rates = true;
+	bool ret = true;
+	// up to maxlimit rates added to rates vector
 
-			auto item_list = reader[L"items"].GetArray();
+	while (retrieve_rates) {
```

Comment:
This can create an infinite loop if `ret` gets flipped to false. (tested with "Tonopah Irrigation District")
I'd recommend changing this to `while (retrieve_rates && ret)`